### PR TITLE
DEPR: Deprecate is_copy (#18801)

### LINKED ISSUE #18801: DEPR: is_copy
this has always been an internal attribute. We can simply replace by ``._is_copy`` and provide a deprecation warning on the property.

### PR DESCRIPTION
**Pull request by cbertinato**
Not all tests for this branch on the fork are passing, but these are the same tests that fail for the clean master and appear to be unrelated to the changes.

References to is_copy in tests were changed to the internal attribute _is_copy assuming that, though is_copy is not meant to be a public attribute, checking the state of this attribute is necessary to check consistent internal behavior.

Closes #18801.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -132,10 +132,22 @@ def __init__(self, data, axes=None, copy=False, dtype=None,
...
 
-        object.__setattr__(self, 'is_copy', None)
+        object.__setattr__(self, '_is_copy', None)
         object.__setattr__(self, '_data', data)
         object.__setattr__(self, '_item_cache', {})
 
+    @property
+    def is_copy(self):
+        warnings.warn("Attribute 'is_copy' will be removed in a future "
+                      "version.", FutureWarning, stacklevel=2)
```
Comment (gfyoung) [inline pandas/core/generic.py]: Say : "is deprecated and will removed..."

Comment (gfyoung) [reply pandas/core/generic.py]: Also, add a test for this deprecation (both for getter and setter).

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a whatsnew note in the deprecation section & a test (as @gfyoung indicates); you can put this in pandas/tests/indexing/test_chainging_and_caching.py

**Comment by cbertinato** [comment]
Shall I squash these commits into one and resubmit, or can we do it on the merge?

**Comment by jreback** [comment]
no need to squash

will be auto done on merge

**Comment by cbertinato** [comment]
I am by no means in a rush, but just wanted to check whether this is waiting on anything else. I believe that the requested changes have been made.

**Comment by cbertinato** [comment]
Roger that!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @cbertinato

**Comment by topper-123** [comment]
When tab-completing in iPython, the ``FutureWarning`` for ``is_copy`` pops up.

I believe that ``'is_copy'`` should be added to ``_deprecations``

**Comment by jreback** [comment]
yep that is right - PR?

**Comment by cbertinato** [comment]
PR #18922

**Pull request merged**